### ISSUE TITLE
!!merge overwrites existing keys

### ISSUE THREAD

**Issue by majewsky** [mikefarah/yq#2110]
**Describe the bug**

Upon switching from python-yq to go-yq, I discovered that some existing YAML files are interpreted differently. Specifically, the `<<` merge key is handled differently by go-yq from what [the spec](https://yaml.org/type/merge.html) suggests. The spec says:

> If the value associated with the key is a single mapping node, each of its key/value pairs is inserted into the current mapping, unless the key already exists in it.

As shown in the example below (specifically, in the object called "ellipse"), go-yq appears to overwrite existing keys during merging. The object called "egg" demonstrates a workaround, in which the merge key is placed before all other keys to avoid this bug.

Version of yq: 4.44.2
Operating system: Arch Linux
Installed via: System package (pacman)

**Input Yaml**
```yaml
objects:
  - &circle
    name: circle
    shape: round
  - name: ellipse
    !!merge <<: *circle
  - !!merge <<: *circle
    name: egg
```

**Command**
The command you ran:
```
yq -o json input.yaml
```

**Actual behavior**

```yaml
{
  "objects": [
    {
      "name": "circle",
      "shape": "round"
    },
    {
      "name": "circle",
      "shape": "round"
    },
    {
      "shape": "round",
      "name": "egg"
    }
  ]
}
```

**Expected behavior**

```yaml
{
  "objects": [
    {
      "name": "circle",
      "shape": "round"
    },
    {
      "name": "ellipse",
      "shape": "round"
    },
    {
      "shape": "round",
      "name": "egg"
    }
  ]
}
```

**Comment by mikefarah** [comment]
Ooh interesting 🤔  I don't know how I missed that. It's a pity they didn't have an example like yours, where the merge key is after the key values - I didn't think of it 😮‍💨 

I'm a little worried that if I change this behavior now it would break a bunch of data pipelines for current users - I don't think this bug can be safely fixed. 

I think that this is probably one of the reasons the merge key has been removed from the 1.2 spec :/ without reading the spec super clearly it's easy to have different expectations on how it would work :(

**Comment by LarsStegman** [comment]
This is definitely a bug that needs to be fixed. I just ran into it and couldn't understand what was happening for hours. The behaviour is wrong, given the YAML standard.

**Comment by mikefarah** [comment]
I've added a flag `--yaml-fix-merge-anchor-to-spec` which will be available in the next release. This will first default to false to maintain existing behavior (with a warning log).

Then later in the year I'll default it to true, followed by its removal ~ 2026.

**Comment by stevenwdv** [comment]
@mikefarah I think this can be closed now?

**Comment by mikefarah** [comment]
Yep 👍🏼 , fixed in [v4.47.1](https://github.com/mikefarah/yq/releases/tag/v4.47.1)

**Comment by Skaronator** [comment]
> Then later in the year I'll default it to true, followed by its removal ~ 2026.

@mikefarah just a reminder, I would say the flag can be enabled by default :)
